File: 10) Remove K digits/remKdig.cpp
#include <iostream>
#include <string>
#include <stack>
using namespace std;

string removeKDigits(string num, int k) {
    int n = num.size();
    stack<char> s;

    for (char digit : num) {
        while (k > 0 && !s.empty() && s.top() > digit) {
            s.pop();
            k--;
        }
        if (!s.empty() || digit != '0') {
            s.push(digit);
        }
    }

    while (k > 0 && !s.empty()) {
        s.pop();
        k--;
    }

    string result;
    while (!s.empty()) {
        result = s.top() + result;
        s.pop();
    }

    return result.empty() ? "0" : result;
}

int main() {
    string num;
    int k;

    cout << "Enter num: ";
    cin >> num;
    cout << "Enter k: ";
    cin >> k;

    string result = removeKDigits(num, k);
    cout << result << endl;

    return 0;
}

File: 10) Remove K digits/remKdig.py
def remove_k_digits(num, k):
    stack = []

    for digit in num:
        while k > 0 and stack and stack[-1] > digit:
            stack.pop()
            k -= 1
        if stack or digit != '0':
            stack.append(digit)

    while k > 0 and stack:
        stack.pop()
        k -= 1

    result = ''.join(stack)
    return result if result else '0'

num = input("Enter num: ")
k = int(input("Enter k: "))

result = remove_k_digits(num, k)
print(result)